### ISSUE TITLE
ClientWebSocket Dispose AV's while operations still in-flight

### ISSUE THREAD

**Issue by KKhurin** [dotnet/corefx#6142]
WCF stress hits an AV while inside an async call to SendAsync on a web socket that is already disposed. 

~~~
0:023> k
00  coreclr!MarshalNative::GCHandleInternalGet [f:\dd\ndp\clr\src\vm\marshalnative.cpp @ 731]
01  mscorlib_ni!System.Runtime.InteropServices.GCHandle.get_Target()+0x1e 
02  System_Net_WebSockets_Client!System.Net.WebSockets.WinHttpWebSocket.SendAsync
03  System_Net_WebSockets_Client!System.Net.WebSockets.ClientWebSocket.SendAsync
04  System_Private_ServiceModel!System.ServiceModel.Channels.WebSocketTransportDuplexSessionChannel+WebSocketStream+<WriteAsync>
05  mscorlib_ni!System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start
06  System_Private_ServiceModel!System.ServiceModel.Channels.WebSocketTransportDuplexSessionChannel+WebSocketStream.WriteAsync
07  System_Private_ServiceModel!System.ServiceModel.Channels.TimeoutStream+<WriteAsync>d__10.MoveNext()
08  mscorlib_ni!System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start
09  System_Private_ServiceModel!System.ServiceModel.Channels.TimeoutStream.WriteAsync
0a  System_Private_ServiceModel!System.ServiceModel.Channels.TimeoutStream+<WriteAsyncInternal> 
0b  mscorlib_ni!System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start
0c  System_Private_ServiceModel!System.ServiceModel.Channels.TimeoutStream.WriteAsyncInternal
0d  System_Private_ServiceModel!System.ServiceModel.Channels.TimeoutStream.Write 
0e  System_Private_DataContractSerialization_ni!System.Xml.XmlStreamNodeWriter.WriteBytes
0f  System_Private_DataContractSerialization_ni!System.Xml.XmlBinaryNodeWriter.WriteBase64Text 
10  System_Private_DataContractSerialization_ni!System.Xml.XmlBaseWriter.WriteBase64(Byte[], Int32, Int32)+0x18d 
11  System_Private_ServiceModel!System.ServiceModel.Dispatcher.StreamFormatter+StreamFormatterHelper.WriteValue
12  System_Private_ServiceModel!System.ServiceModel.Dispatcher.StreamFormatter.Serialize
13  System_Private_ServiceModel!System.ServiceModel.Dispatcher.OperationFormatter.SerializeBodyContents
14  System_Private_ServiceModel!System.ServiceModel.Dispatcher.OperationFormatter+OperationFormatterMessage+OperationFormatterBodyWriter.OnWriteBodyContents
15  System_Private_ServiceModel!System.ServiceModel.Channels.BodyWriterMessage.OnWriteBodyContents
16  System_Private_ServiceModel!System.ServiceModel.Channels.Message.OnWriteMessage
17  System_Private_ServiceModel!System.ServiceModel.Channels.BinaryMessageEncoderFactory+BinaryMessageEncoder.WriteMessage
18  System_Private_ServiceModel!System.ServiceModel.Channels.WebSocketTransportDuplexSessionChannel.OnSendCore
19  System_Private_ServiceModel!System.ServiceModel.Channels.TransportDuplexSessionChannel.OnSend
1a  System_Private_ServiceModel!System.ServiceModel.Channels.OutputChannel.Send
1b  System_Private_ServiceModel!System.ServiceModel.Dispatcher.DuplexChannelBinder+DuplexRequestContext.OnReply
1c  System_Private_ServiceModel!System.ServiceModel.Channels.RequestContextBase.Reply
1d  System_Private_ServiceModel!System.ServiceModel.Dispatcher.ImmutableDispatchRuntime.Reply
1e  System_Private_ServiceModel!System.ServiceModel.Dispatcher.ImmutableDispatchRuntime.ProcessMessage9
1f  System_Private_ServiceModel!System.ServiceModel.Dispatcher.ImmutableDispatchRuntime.ProcessMessage8
20  System_Private_ServiceModel!System.ServiceModel.Dispatcher.MessageRpc.Process
21  System_Private_ServiceModel!System.ServiceModel.Dispatcher.MessageRpc+Wrapper.Resume
22  System_Private_ServiceModel!System.ServiceModel.Dispatcher.ThreadBehavior.CleanThreadCallback
23  mscorlib_ni!System.Threading.Tasks.Task.Execute()
~~~

**Comment by davidsh** [comment]
@KKhurin Can you please post repro instructions? Thx.

**Comment by davidsh** [comment]
Fixed with #6500.
